enclave_build/src/lib.rs

```diff
@@ -12,14 +12,63 @@ mod yaml_generator;
 use aws_nitro_enclaves_image_format::defs::{EifBuildInfo, EifIdentityInfo, EIF_HDR_ARCH_ARM64};
 use aws_nitro_enclaves_image_format::utils::identity::parse_custom_metadata;
 use aws_nitro_enclaves_image_format::utils::{EifBuilder, SignEnclaveInfo};
-use docker::DockerUtil;
+use docker::{DockerError, DockerUtil};
 use serde_json::json;
 use sha2::Digest;
 use std::collections::BTreeMap;
-use yaml_generator::YamlGenerator;
+use thiserror::Error;
+use yaml_generator::{YamlGenerator, YamlGeneratorError};
 
 pub const DEFAULT_TAG: &str = "1.0";
 
+#[derive(Debug, Error)]
+pub enum EnclaveBuildError {
+    #[error("Docker error: `{0:?}`")]
+    DockerError(DockerError),
+    #[error("Invalid path: `{0}`")]
+    PathError(String),
+    #[error("Image pull error")]
+    ImagePullError,
+    #[error("Linuxkit error: `{0}`")]
+    LinuxKitError(String),
+    #[error("File operation error")]
+    FileError(std::io::Error),
+    #[error("Ramfs error: `{0:?}`")]
+    RamfsError(YamlGeneratorError),
+    #[error("Signature error: `{0}`")]
+    SignError(String),
+    #[error("Metadata error: `{0}`")]
+    MetadataError(String),
+    #[error("Unsupported architecture")]
+    UnsupportedArchError,
+    #[error("Getting image detail failed: `{0}`")]
+    ImageDetailError(String),
+    #[error("Container image operation failed")]
+    ImageOperationError,
+    #[error("Failed to convert image to EIF")]
+    ImageConvertError,
+    #[error("Hashing error: `{0}`")]
+    HashingError(String),
+    #[error("Serde error: `{0:?}`")]
+    SerdeError(serde_json::Error),
+    #[error("Credentials error")]
+    CredentialsError,
+    #[error("Image cache initialization error")]
+    CacheInitError,
+    #[error("Cache store operation failed: `{0}`")]
+    CacheStoreError(String),
+    #[error("Config error")]
+    ConfigError,
+    #[error("Manifest error")]
+    ManifestError,
+    #[error("Failed to extract expressions from image: `{0}`")]
+    ExtractError(String),
+    #[error("EIF build error: `{0}`")]
+    OtherError(String),
+}
+
```

Comment:
Removed specific errors from each file. Centralized the error types into `EnclaveBuildError` without the need to implement the `Display` trait for each of them as we use `thiserror`. Modified each occurrence of these errors.